### PR TITLE
[TF-35] Allow custom IP and port for service

### PR DESCRIPTION
**Pull request by MattFiler**
## Summary

Adds the ability to set `ServiceIP` and `ServicePort` in ServiceConfig.json which is then respected by the service & Unity applications and tooling.

https://ultrahaptics.atlassian.net/browse/TF-35

### Tests Added

Created test [TF-T454 (1.0)](https://ultrahaptics.atlassian.net/projects/TF?selectedItem=com.atlassian.plugins.atlassian-connect-plugin:com.kanoah.test-manager__main-project-page#!/testCase/TF-T454)

## Contributor Tasks

_These tasks are for the pull request creator to tick off._

- [ ] PO review (optional depending on work type)
- [ ] XDR review (optional depending on work type)
- [ ] QA review (or another developer if no QA is available)
- [x] Ensure documentation requirements are met e.g., public API is commented
- [x] Relevant changelogs have been updated with user-visible changes
    - [x] [TouchFree Windows](/ultraleap/touchfree/blob/-/CHANGELOG-windows.md)
    - [ ] [TouchFree BrightSign](/ultraleap/touchfree/blob/-/CHANGELOG-brightsign.md)
    - [ ] [TouchFree Web Tooling](/ultraleap/touchfree/blob/-/TF_Tooling_Web/CHANGELOG.md)
- [ ] Consider any licensing/other legal implications e.g., notices required

If there is an associated JIRA issue:
- [x] Include a link to the JIRA issue in the summary above
- [x] Make sure the fix version on the issue is set correctly

## Reviewer Tasks

_Add any instructions or tasks for the reviewer such as specific test considerations before this can be merged._

- [x] Developer testing
- [x] Code reviewed
- [ ] Non-code assets reviewed
- [x] Documentation reviewed - includes checking documentation requirements are met and not missing e.g., public API is commented

### CODE REVIEW

File: TF_Service_dotNet/TouchFree_Service/Program.cs [hunk 1]
```diff
@@ -24,7 +31,7 @@ static void Main(string[] args)
                 .ConfigureWebHostDefaults(webBuilder =>
                 {
                     webBuilder.UseStartup<Startup>();
-                    webBuilder.UseUrls("http://localhost:9739");
+                    webBuilder.UseUrls("http://" + config.ServiceIP + ":" + config.ServicePort);
```
Comment (GabrielHomewoodUL) [inline TF_Service_dotNet/TouchFree_Service/Program.cs]: Can you change this one to use string interpolation:
`$"http://{config.ServiceIP}:{config.ServicePort}"`

Comment (MattFiler) [reply TF_Service_dotNet/TouchFree_Service/Program.cs:34]: Sure - would you rather this on the Unity side too?

**Comment by JordanBerni-UL** [comment]
Dev-QA approved!

**Comment by MattFiler** [comment]
Docs updated over [here](https://gitlab.ultrahaptics.com/marcom/ultraleap-api-docs/-/merge_requests/373) 👍

**Review by JordanBerni-UL — APPROVED** [review]

**Pull request merged**